### ISSUE TITLE
False positive E1101

### ISSUE THREAD

**Issue by Anvolne** [PyCQA/pylint#3331]
First of all, if this exact problem has already been reported, I am sorry. But I couldn't find anything resembling this, so here it is:

### Steps to reproduce
For this file, when run with `pylint -d C,R test.py`, pylint incorrectly infers `Sub` as `Base`:
```py
class Base:
    def __enter__(self):
        return self

    def __exit__(self, *err):
        pass


class Base2:
    def base_func2(self, something):
        return something


class Sub(Base):
    def some_func(self, string):
        print(string)


class Sub2(Base, Base2):
    def __init__(self, arg1, arg2):
        with Sub() as sub:
            sub.some_func(arg1)
            sub.some_func(arg2)
        self.sub = sub
```

### Current behavior
Output:
```
************* Module test
test.py:22:12: E1101: Instance of 'Base' has no 'some_func' member (no-member)
test.py:23:12: E1101: Instance of 'Base' has no 'some_func' member (no-member)

-------------------------------------------------------------------
Your code has been rated at 4.12/10 (previous run: -2.94/10, +7.06)
```

### Expected behavior
No errors

### pylint --version output
```
$ pylint --version
pylint 2.4.4
astroid 2.3.3
Python 3.8.1 (default, Dec 21 2019, 20:57:38) 
[GCC 9.2.0]
```

Thank you in advance

**Comment by AWhetter** [comment]
Thanks for the report. This has already been fixed in astroid for #3157.
